src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerService.cs

```
@@ -0,0 +1,247 @@
+                }
+
+                await requestContext.SendResult(result);
+            }
+            catch (Exception e)
+            {
+                await requestContext.SendError(e);
+            }
+        }
+
+        /// <summary>
+        /// Handle request to stop a profiling session
+        /// </summary>
+        internal async Task HandleStopProfilingRequest(StopProfilingParams parameters, RequestContext<StopProfilingResult> requestContext)
+        {
+            try
+            {
+                monitor.StopMonitoringSession(parameters.SessionId);
+                await requestContext.SendResult(new StopProfilingResult());
+            }
+            catch (Exception e)
+            {
+                await requestContext.SendError(e);
+            } 
+        }
+
+        /// <summary>
+        /// Starts a new profiler session for the provided connection
+        /// </summary>
+        internal ProfilerSession StartSession(ConnectionInfo connInfo)
+        {
+            // create a new XEvent session and Profiler session
+            var xeSession = this.XEventSessionFactory.CreateXEventSession(connInfo);
+            var profilerSession = new ProfilerSession()
+            {
+                SessionId = Guid.NewGuid().ToString(),
+                XEventSession = xeSession
+            };
+
+            // start monitoring the profiler session
+            monitor.StartMonitoringSession(profilerSession);
+
+            return profilerSession;
+        }
+
+        /// <summary>
+        /// Create a new XEvent sessions per the IXEventSessionFactory contract
+        /// </summary>
+        public IXEventSession CreateXEventSession(ConnectionInfo connInfo)
+        {
+            var sqlConnection = ConnectionService.OpenSqlConnection(connInfo);
+            SqlStoreConnection connection = new SqlStoreConnection(sqlConnection);
+            Session session = ProfilerService.GetOrCreateSession(connection, "Profiler");
```

>GetOrCreateSession [](start = 46, length = 18)

this method should probably return an error message ( inside the session object or other ways) for when creating session fails. If session is returned as null, we don't know why

---

I think that if something goes wrong we'll raise an exception.  The error message would be constructed in the outer-most request handler.